GPU/Workflow/src/GPUWorkflowSpec.cxx

```diff
@@ -376,6 +406,55 @@ DataProcessorSpec getGPURecoWorkflowSpec(gpuworkflow::CompletionPolicyData* poli
     };
     ic.services().get<CallbackService>().set(CallbackService::Id::Stop, printTiming);
 
+    auto finaliseCCDB = [processAttributes](ConcreteDataMatcher& matcher, void* obj) {
+      LOGP(info, "checking for newer object....");
+      CalibdEdxContainer* dEdxCalibContainer = processAttributes->dEdxCalibContainer.get();
+
+      bool updateCalibs = false;
+      if (matcher == ConcreteDataMatcher(gDataOriginTPC, "PADGAINFULL", 0)) {
+        LOGP(info, "Updating gain map from CCDB");
+        const auto* gainMap = static_cast<o2::tpc::CalDet<float>*>(obj);
+        const float minGain = 0;
+        const float maxGain = 2;
+        processAttributes->dEdxCalibContainer.get()->setGainMap(*gainMap, minGain, maxGain);
+        updateCalibs = true;
+      } else if (matcher == ConcreteDataMatcher(gDataOriginTPC, "PADGAINRESIDUAL", 0)) {
+        LOGP(info, "Updating residual gain map from CCDB");
+        const auto* gainMapResidual = static_cast<std::unordered_map<string, o2::tpc::CalDet<float>>*>(obj);
+        const float minResidualGain = 0.7f;
+        const float maxResidualGain = 1.3f;
+        processAttributes->dEdxCalibContainer.get()->setGainMapResidual(gainMapResidual->at("GainMap"), minResidualGain, maxResidualGain);
+        updateCalibs = true;
+      } else if (matcher == ConcreteDataMatcher(gDataOriginTPC, "PADTHRESHOLD", 0)) {
+        LOGP(info, "Updating threshold map from CCDB");
+        const auto* thresholdMap = static_cast<std::unordered_map<string, o2::tpc::CalDet<float>>*>(obj);
+        processAttributes->dEdxCalibContainer.get()->setZeroSupresssionThreshold(thresholdMap->at("ThresholdMap"));
+        updateCalibs = true;
+      } else if (matcher == ConcreteDataMatcher(gDataOriginTPC, "TOPOLOGYGAIN", 0) && !(dEdxCalibContainer->isTopologyCorrectionSplinesSet())) {
+        LOGP(info, "Updating Q topology correction from CCDB");
+        const auto* topologyCorr = static_cast<o2::tpc::CalibdEdxTrackTopologyPolContainer*>(obj);
+        CalibdEdxTrackTopologyPol calibTrackTopology;
+        calibTrackTopology.setFromContainer(*topologyCorr);
+        dEdxCalibContainer->setPolTopologyCorrection(calibTrackTopology);
+        updateCalibs = true;
+      } else if (matcher == ConcreteDataMatcher(gDataOriginTPC, "TIMEGAIN", 0)) {
+        LOGP(info, "Updating residual gain correction from CCDB");
+        const auto* residualCorr = static_cast<o2::tpc::CalibdEdxCorrection*>(obj);
+        dEdxCalibContainer->setResidualCorrection(*residualCorr);
+        updateCalibs = true;
+      }
+
+      if (updateCalibs) {
+        // updating the calibration object
+        GPUCalibObjectsConst newTopologyCalib;
+        newTopologyCalib.dEdxCalibContainer = dEdxCalibContainer;
+        auto& tracker = processAttributes->tracker;
+        tracker->UpdateCalibration(newTopologyCalib);
+      }
```

Comment:
I'm not sure this is the best place to trigger the `updateCalibration`, since if several calibrations are updated in the the same TF, this will be called multiple times. Perhaps the `updateCalibs` should be propagated to the `processingFct` and `UpdateCalibration` called just before `RunTracking`.